File: Java/src/main/java/com/gildedrose/Items/ConjuredItem.java
package com.gildedrose.Items;

import com.gildedrose.Item;

public class ConjuredItem extends GildedItem {
    public ConjuredItem(Item item) {
        super(item);
    }

    @Override
    public void updateItemQuality() {
        decrementItemQuality();
        decrementItemQuality();
        item.sellIn--;
        if (item.sellIn < 0) {
            decrementItemQuality();
            decrementItemQuality();
        }
    }
}

File: go/gildedrose/conjured_item.go
package gildedrose

type ConjuredItem struct {
	*Item
}

func (item *ConjuredItem) itemUpdateStrategy() {
	item.decQuality()
	item.decQuality()
	item.SellIn--
	if item.SellIn < 0 {
		item.decQuality()
		item.decQuality()
	}
}
